### PR TITLE
Add attribute scope validation

### PR DESCRIPTION
**Pull request by nalquas**
This PR adds attribute scope validation functionality to validate_ontology, as well as lots of tests to check some edge cases.

### CODE REVIEW

File: raillabel_providerkit/validation/validate_ontology/_ontology_classes/_ontology.py [hunk 1]
```diff
@@ -50,6 +52,59 @@ def _check_class_validity(self, scene: raillabel.Scene) -> None:
...
+    ) -> list[Issue]:
+        errors: list[Issue] = []
+        checked_attributes_by_object_type: dict[str, list[str]] = {}
+
+        for annotation_with_metadata in annotations_with_metadata:
+            object_type = annotation_with_metadata.object_type
+
+            if object_type not in self.classes:
+                continue
+            object_class = self.classes[object_type]
```
Comment (unexcellent) [inline raillabel_providerkit/validation/validate_ontology/_ontology_classes/_ontology.py:66]: Using `object_type` and `object_class` as separate variables might be confusing here because type and class are sometimes used interchangeably in raillabel. Consider renaming `object_type` to `object_type_name` or something like this.

File: raillabel_providerkit/validation/validate_ontology/_ontology_classes/_ontology.py [hunk 2]
```diff
@@ -50,6 +52,59 @@ def _check_class_validity(self, scene: raillabel.Scene) -> None:
...
+        self, annotations_with_metadata: list[_AnnotationWithMetadata]
+    ) -> list[Issue]:
+        errors: list[Issue] = []
+        checked_attributes_by_object_type: dict[str, list[str]] = {}
+
+        for annotation_with_metadata in annotations_with_metadata:
+            object_type = annotation_with_metadata.object_type
+
+            if object_type not in self.classes:
+                continue
```
Comment (unexcellent) [inline raillabel_providerkit/validation/validate_ontology/_ontology_classes/_ontology.py:65]: It is not immediately obvious to me why the iteration can be skipped here (because this is handled by a separate issue?). Can you add a comment explaining why it is skipped?

**Review by unexcellent — APPROVED** [review]

File: raillabel_providerkit/validation/validate_ontology/_ontology_classes/_ontology.py (hunk 2 above)
Comment (nalquas) [reply raillabel_providerkit/validation/validate_ontology/_ontology_classes/_ontology.py:66]: Added a note
